src/api/web/endpoint/package.ts

```diff
@@ -100,7 +100,13 @@ function addPackageWebApi(route: Router, storage: IStorageHandler, auth: IAuth,
         }
 
         res.set(HEADER_TYPE.CONTENT_TYPE, HEADERS.TEXT_PLAIN);
-        next(parseReadme(info.name, info.readme));
+        let readMeHtml = parseReadme(info.name, info.readme);
+        if ('string' === typeof readMeHtml && 'string' === typeof req.headers.referer) {
+          readMeHtml = readMeHtml.replace(
+            /href="#/gi,
+            `href="${(new URL(req.headers.referer)).pathname}#`);
```

Comment:
Instead `req.headers.referer` there is a safest way using `request.get()` http://expressjs.com/en/api.html#req.get